sys/net/gnrc/network_layer/ipv6/gnrc_ipv6.c

```diff
@@ -852,7 +852,6 @@ static void _receive(gnrc_pktsnip_t *pkt)
         ipv6 = gnrc_pktbuf_mark(pkt, sizeof(ipv6_hdr_t), GNRC_NETTYPE_IPV6);
 
         first_ext = pkt;
-        pkt->type = GNRC_NETTYPE_UNDEF; /* snip is no longer IPv6 */
```

Comment:
I checked the implementation of gnrc_pkt_len_upto ([here], line 147). The sum is made before the if(pkt->type == type), so what I see is if type is GNRC_NETTYPE_UNDEF, it will count the payload plus ipv6 hdr.

[here]: http://riot-os.org/api/pkt_8h_source.html